### PR TITLE
designate: Configure resource settings (SOC-9633)

### PR DESCRIPTION
**Pull request by s-t-e-v-e-n-k**
Designate requires a project and a responsible person contact for any
zones it creates, add them into the barclamp configuration with the
first migratation for designate.

### CODE REVIEW

File: chef/data_bags/crowbar/migrate/designate/301_add_resources.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+def upgrade(template_attrs, template_deployment, attrs, deployment)
+  attrs["resource_email"] = ""
```
Comment (nicolasbock) [inline chef/data_bags/crowbar/migrate/designate/301_add_resources.rb]: We usually use this kind of pattern:
```
attrs["resource_email"] = "" unless attrs.key? "resource_email"
```

File: chef/data_bags/crowbar/migrate/designate/301_add_resources.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+  attrs["resource_project_id"] = ""
+  return attrs, deployment
+end
+
+def downgrade(template_attrs, template_deployment, attrs, deployment)
+  attrs.delete("resource_email", "resource_project_id")
```
Comment (nicolasbock) [inline chef/data_bags/crowbar/migrate/designate/301_add_resources.rb]: Similarly here:
```
attrs.delete("resource_email", "resource_project_id") unless template_attrs.key? "resource_email" or template_attrs.key? "resource_project_id"
```

File: chef/data_bags/crowbar/migrate/designate/301_add_resources.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+def upgrade(template_attrs, template_deployment, attrs, deployment)
+  attrs["resource_email"] = "" unless attrs.keys? "resource_email"
+  attrs["resource_project_id"] = "" unless attrs.keys? "resource_project_id"
+  return attrs, deployment
+end
+
+def downgrade(template_attrs, template_deployment, attrs, deployment)
+  attrs.delete("resource_email") unless template_attrs.keys? "resource_email"
+  unless template_attrs.keys? "resource_project_id"
```
[the remark this answers is not in the document]
Comment (matthewoliver) [reply chef/data_bags/crowbar/migrate/designate/301_add_resources.rb]: if you did something like: http://pastebin.nue.suse.com/20734/src

You should appease the angry dog.

File: chef/data_bags/crowbar/template-designate.json [hunk 4]
```diff
@@ -8,6 +8,8 @@
       "keystone_instance": "none",
       "database_instance": "none",
       "rabbitmq_instance": "none",
+      "resource_email": "",
+      "resource_project": "",
```
Comment (dirkmueller) [inline chef/data_bags/crowbar/template-designate.json]: can we use a reasonable default? "service" perhaps? "openstack"? not sure which of the pregenerated projects would be a good one

File: crowbar_framework/config/locales/designate/en.yml [hunk 5]
```diff
@@ -21,3 +21,7 @@ en:
         rabbitmq_instance: 'RabbitMQ'
         database_instance: 'Database Instance'
         keystone_instance: 'Keystone'
+        resource_email: 'E-mail address of hostmaster'
+        resource_project: 'Project that will own created resources'
```
Comment (dirkmueller) [inline crowbar_framework/config/locales/designate/en.yml]: not used? or did you forget to git add your change in edit_attributes.haml?

File: chef/data_bags/crowbar/migrate/designate/301_add_resources.rb [hunk 6]
```diff
@@ -0,0 +1,11 @@
+def upgrade(template_attrs, template_deployment, attrs, deployment)
+  attrs["resource_email"] = "" unless attrs.keys? "resource_email"
```
Comment (toabctl) [inline chef/data_bags/crowbar/migrate/designate/301_add_resources.rb]: Shouldn't that be `attrs["resource_email"] = template_attrs["resource_email"] unless ......` to use the defaults?

File: chef/data_bags/crowbar/migrate/designate/301_add_resources.rb [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+  attrs["resource_project"] = "" unless attrs.keys? "resource_project"
```
Comment (toabctl) [inline chef/data_bags/crowbar/migrate/designate/301_add_resources.rb]: same here

File: chef/data_bags/crowbar/migrate/designate/301_add_resources.rb [hunk 8]
```diff
@@ -0,0 +1,11 @@
+def upgrade(template_attrs, template_deployment, attrs, deployment)
+  attrs["resource_email"] = template_attrs["resource_email"] unless attrs.keys? "resource_email"
+  attrs["resource_project"] = template_attrs["resource_project"] unless attrs.keys? "resource_project"
```
[the remark this answers is not in the document]
Comment (matthewoliver) [reply chef/data_bags/crowbar/migrate/designate/301_add_resources.rb]: *sigh* thanks hound, and yeah if you make it an:

  ```
unless attrs.keys? "resource_project"
  attrs["resource_project"] = template_attrs["resource_project"]
```

Hound will complain that there is only one statement, so you should use the "statement unless ..." syntax.

The only other way around it, without just ignoring hound (which I'm not against) is something like:

  ```
resource_project = template_attrs["resource_project"]
attrs["resource_project"] = resource_project unless attrs.keys? "resource_project"

```
As `resource_project` should just be defined as nil if the key doesn't exist.. though that isn't pretty.

Either way, I'm happy either way. IE talk dirk into merging it anyway for trying a different way of shutting up hound.. good luck :P

File: crowbar_framework/config/locales/designate/en.yml (hunk 5 above)
Comment (dirkmueller) [reply crowbar_framework/config/locales/designate/en.yml]: fixed now

**Pull request merged**